php/libraries/Candidate.class.inc

```diff
@@ -699,52 +699,92 @@ class Candidate
     }
 
     /**
+    * Generates a new id in the format defined by the settings
+    *
+    * @param string   $name                  The name to display on exception
+    * @param array    $settings              The xml settings
+    * @param array    $static_dict           Example: array("siteAbbrev"=>"DCC")
+    * @param function $is_colliding_callback `bool ($needle)`
+    *
+    * @return string The id
+    */
+    private static function _generateId(
+        $name,
+        $settings,
+        $static_dict,
+        $is_colliding_callback
+    ) {
+        $generator = Candidate::idGeneratorFromXMLConfig(
+            $settings,
+            $static_dict
+        );
+
+        $is_sequential = (
+            isset($settings["generation"])
+            && $settings["generation"] == "sequential"
+        );
+
+        $result = null;
+        if ($is_sequential) {
```

Comment:
Adding that strict equality doesn't change the semantics so I'm okay with just throwing it in.